frontend/src/components/design-system/input/ButtonComponent.vue

```diff
@@ -0,0 +1,270 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+
+import ButtonText from '@/components/design-system/typography/ButtonText.vue';
+
+export interface Props {
+  tag?: 'button' | 'a';
+  href?: string;
+  kind: 'link' | 'outlined' | 'filled' | 'answer';
+  size?: 'medium' | 'small';
+  color?: 'neutral' | 'primary' | 'secondary';
+  selected?: boolean;
```

Comment:
Všiml jsem si toho až tady, ale zpětně si uvědomuji, že to tu je mnohokrát. Pokud proměnná je typu boolean, tak good practice je ji pojmenovat s 'is' nabo 'has' na začátku. V tomto případě tedy `isSelected`, v kódu pak například bude `isSelected = true`. Jedná se čistě o přidání informace k proměnné. 

Pokud se něco jmenuje `selected`, tak bych čekal, že hodnota bude nějaký index, nebo hodnota obecně, odkazující někam do objektu. Například číslo řádku nebo pole id. Porovnej s `selected.push(user.id)`.

Reply:
Whatever, asi může být, píšu si do seznamu k refactoringu upravím pak hromadně tedy.